### PR TITLE
run_swag.py should use AdamW

### PR DESCRIPTION
**Pull request by jeff-da**
run_swag.py doesn't compile currently, BertAdam is removed (per readme).

### CODE REVIEW

File: examples/single_model_scripts/run_swag.py [hunk 1]
```diff
@@ -467,10 +467,13 @@ def main():
                 if (step + 1) % args.gradient_accumulation_steps == 0:
                     if args.fp16:
```
Comment (thomwolf) [inline examples/single_model_scripts/run_swag.py]: We don't need this seperate adjustement of learning rate for fp16 anymore with the schedulers

File: examples/single_model_scripts/run_swag.py [hunk 2]
```diff
@@ -428,10 +428,10 @@ def main():
             warmup_linear = WarmupLinearSchedule(warmup=args.warmup_proportion,
                                                  t_total=num_train_optimization_steps)
         else:
-            optimizer = BertAdam(optimizer_grouped_parameters,
-                                 lr=args.learning_rate,
-                                 warmup=args.warmup_proportion,
-                                 t_total=num_train_optimization_steps)
+            optimizer = AdamW(optimizer_grouped_parameters, lr=args.learning_rate)
+            scheduler = WarmupLinearSchedule(optimizer,
```
Comment (thomwolf) [inline examples/single_model_scripts/run_swag.py]: The scheduler should be created also if we are in fp16.
The fp16 optimizer should now be created like in the `run_glue` example where there is no distinction between fp16 and normal operation.

**Comment by thomwolf** [comment]
Added a few comments. If you take a look at the `run_glue` and `run_squad` examples, you'll see they are much simpler now in term of optimizer setup. This example could take advantage of the same refactoring if you want to give it a look!

**Comment by thomwolf** [comment]
Thanks for this @jeff-da, we'll close this PR in favor of #1004 for now.
Feel free to re-open if there are other things you would like to change.
